aixplain/modules/pipeline/asset.py

```diff
@@ -152,41 +165,42 @@ def run(
         timeout: float = 20000.0,
         wait_time: float = 1.0,
         **kwargs,
-    ) -> Dict:
-        """Runs a pipeline call.
-
-        Args:
-            data (Union[Text, Dict]): link to the input data
-            data_asset (Optional[Union[Text, Dict]], optional): Data asset to be processed by the pipeline. Defaults to None.
-            name (Text, optional): ID given to a call. Defaults to "pipeline_process".
-            timeout (float, optional): total polling time. Defaults to 20000.0.
-            wait_time (float, optional): wait time in seconds between polling calls. Defaults to 1.0.
-            kwargs: A dictionary of keyword arguments. The keys are the argument names
-
-        Returns:
-            Dict: parsed output from pipeline
-        """
+    ) -> PipelineResponse:
         start = time.time()
         try:
             response = self.run_async(data, data_asset=data_asset, name=name, **kwargs)
-            if response["status"] == "FAILED":
+            if response["status"] == Status.FAILED: 
                 end = time.time()
-                response["elapsed_time"] = end - start
-                return response
+                return PipelineResponse(
+                    status=Status.FAILED,
+                    error={"error": response.error, "status": "ERROR"},
+                    elapsed_time=end - start,
+                    **kwargs
+                )
             poll_url = response["url"]
+            polling_response = self.__polling(poll_url, name=name, timeout=timeout, wait_time=wait_time) 
             end = time.time()
-            response = self.__polling(poll_url, name=name, timeout=timeout, wait_time=wait_time)
+            status = Status(polling_response["status"])
```

Comment:
Regarding above my comment, for example here the value of "status" not seems to a be enum. Best to make sure of it.

Reply:
Okay I'll check again